### PR TITLE
Amls 4210 - New incomplete design: Responsible people

### PR DESCRIPTION
**Pull request by slavvco**
Implemented the new incomplete journey for responsible people, updated pages flow and styling.

## Related / Dependant PRs?

n/a

## How Has This Been Tested?

* unit tests
* partial acceptance tests
* manual tests

## Checklist

- [x] Breaking change (fix or feature that would cause existing functionality to change).
- [x] Build passing.
- [x] Unit tests have full coverage.
- [ ] Unit test approach and implementation has been reviewed with QA (testers).
- [x] Requires acceptance test run.
- [x] Appropriate labels added.

### CODE REVIEW

File: test/views/responsiblepeople/your_answersSpec.scala [hunk 1]
```diff
@@ -17,9 +17,12 @@
 package views.responsiblepeople
```
Comment (drewmiley) [inline test/views/responsiblepeople/your_answersSpec.scala]: your_responsible_peopleSpec

File: test/views/responsiblepeople/your_answersSpec.scala [hunk 2]
```diff
@@ -30,21 +33,81 @@ class your_answersSpec extends AmlsSpec with MustMatchers {
...
+
+      doc.getElementById("addResponsiblePerson").text must be(Messages("responsiblepeople.check_your_answers.add"))
+      doc.getElementById("addResponsiblePerson").attr("href") must be(controllers.responsiblepeople.routes.ResponsiblePeopleAddController.get(false).url)
+    }
+
+    "have an incomplete/complete sections with people names displayed and edit/remove links" in new ViewFixture {
+      def view = views.html.responsiblepeople.your_answers(incompleteRpSeq, completeRpSeq)
+
+      doc.getElementById("incomplete-header").text must include(Messages("responsiblepeople.check_your_answers.incomplete"))
+      html must include("Katie Test")
```
Comment (drewmiley) [inline test/views/responsiblepeople/your_answersSpec.scala]: Should check that this text is inside the incomplete section

File: test/views/responsiblepeople/your_answersSpec.scala [hunk 3]
```diff
@@ -30,21 +33,81 @@ class your_answersSpec extends AmlsSpec with MustMatchers {
...
+      doc.getElementById("detail-edit-1").attr("href") must be(controllers.responsiblepeople.routes.PersonNameController.get(2, false, None).url)
+      doc.getElementById("detail-remove-1").attr("href") must be(controllers.responsiblepeople.routes.RemoveResponsiblePersonController.get(2, true).url)
+      doc.getElementById("complete-header").text must include(Messages("responsiblepeople.check_your_answers.complete"))
+      html must include("John Test")
```
Comment (drewmiley) [inline test/views/responsiblepeople/your_answersSpec.scala]: Should check that this is inside the incomplete section

File: app/views/responsiblepeople/your_responsible_people.scala.html [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        <li>@Messages("responsiblepeople.whomustregister.line_4")</li>
+        <li>@Messages("responsiblepeople.whomustregister.line_5")</li>
+    </ul>
+
+    <p><a id="addResponsiblePerson" href="@controllers.responsiblepeople.routes.ResponsiblePeopleAddController.get(false)">@Messages("responsiblepeople.check_your_answers.add")</a></p>
+
+    @if(responsiblePeopleIncomplete.nonEmpty) {
+       <p><h2 id="incomplete-header">@Messages("responsiblepeople.check_your_answers.incomplete")</h2></p>
+        <ul class="list" id="incomplete-rp-list">
+        @responsiblePeopleIncomplete.map { case (rp, index) =>
```
Comment (drewmiley) [inline app/views/responsiblepeople/your_responsible_people.scala.html]: Can use .zipWithIndex to get index

File: app/controllers/responsiblepeople/YourResponsiblePeopleController.scala [hunk 5]
```diff
@@ -21,24 +21,29 @@ import connectors.DataCacheConnector
...
-  val dataCacheConnector: DataCacheConnector
+  def dataCacheConnector: DataCacheConnector
 
   def get() =
       Authorised.async {
         implicit authContext => implicit request =>
           dataCacheConnector.fetch[Seq[ResponsiblePerson]](ResponsiblePerson.key) map {
-            case Some(data) => Ok(your_answers(data))
+            case Some(data) => {
+              val s = ResponsiblePerson.filterWithIndex(data)
```
Comment (drewmiley) [inline app/controllers/responsiblepeople/YourResponsiblePeopleController.scala]: val (completeRP, incompleteRP) = ...
and pass into view.

your_responsible_people(completeRP, incompleteRP)

File: test/models/responsiblepeople/ResponsiblePersonSpec.scala [hunk 6]
```diff
@@ -125,21 +125,22 @@ class ResponsiblePersonSpec extends PlaySpec with MockitoSugar with ResponsibleP
...
             .thenReturn(Some(Seq(completeModelNonUkResidentNonUkPassport.copy(hasAccepted = true))))
 
-          ResponsiblePerson.section(mockCacheMap).call must be(controllers.responsiblepeople.routes.YourAnswersController.get())
+          ResponsiblePerson.section(mockCacheMap).call must be(controllers.responsiblepeople.routes.YourResponsiblePeopleController.get())
         }
       }
 
       "is partially complete" must {
-        "direct the user to the start of the the journey at the correct index for the incomplete item" in {
+        "direct the user to the 'your anwsers' page to show the incomplete items" in {
```
[the remark this answers is not in the document]
Comment (slavvco) [reply test/models/responsiblepeople/ResponsiblePersonSpec.scala]: done

File: app/controllers/responsiblepeople/YourResponsiblePeopleController.scala (hunk 5 above)
Comment (slavvco) [reply app/controllers/responsiblepeople/YourResponsiblePeopleController.scala]: done

File: test/views/responsiblepeople/your_answersSpec.scala (hunk 1 above)
Comment (slavvco) [reply test/views/responsiblepeople/your_answersSpec.scala]: done

File: test/views/responsiblepeople/your_answersSpec.scala [hunk 7]
```diff
@@ -30,21 +33,81 @@ class your_answersSpec extends AmlsSpec with MustMatchers {
...
+  val completeRpSeq = Seq((completeRp1,0),(completeRp2,1))
+  val incompleteRpSeq = Seq((incompleteRp1,2),(incompleteRp2,3))
+
   "What you need View" must {
-    "have correct title, headings and form fields" in new ViewFixture {
+    "have correct title, headings, form fields and bullet list of types of RP's to register displayed" in new ViewFixture {
       val form2 = EmptyForm
 
-      def view = views.html.responsiblepeople.your_answers(Seq(ResponsiblePerson()))
+      def view = views.html.responsiblepeople.your_answers(Seq((ResponsiblePerson(),0)), Seq((ResponsiblePerson(),0)))
```
[the remark this answers is not in the document]
Comment (slavvco) [reply test/views/responsiblepeople/your_answersSpec.scala]: done

File: test/views/responsiblepeople/your_answersSpec.scala (hunk 2 above)
Comment (slavvco) [reply test/views/responsiblepeople/your_answersSpec.scala]: done

File: test/views/responsiblepeople/your_answersSpec.scala (hunk 3 above)
Comment (slavvco) [reply test/views/responsiblepeople/your_answersSpec.scala]: done

File: test/controllers/responsiblepeople/RemoveResponsiblePersonControllerSpec.scala [hunk 8]
```diff
@@ -220,9 +220,9 @@ class RemoveResponsiblePersonControllerSpec extends AmlsSpec
           when(controller.statusService.getStatus(any(), any(), any()))
             .thenReturn(Future.successful(NotCompleted))
 
-          val result = controller.remove(1, false)(request)
+          val result = controller.remove(1, true)(request)
```
[the remark this answers is not in the document]
Comment (slavvco) [reply test/controllers/responsiblepeople/RemoveResponsiblePersonControllerSpec.scala]: unnecessary parameter, removed, refactored controllers etc.

File: public/amls.css [hunk 9]
```diff
@@ -828,6 +828,26 @@ fieldset .panel-indent {
   word-break: break-word;
 }
 
+ul#complete-rp-list {
```
[the remark this answers is not in the document]
Comment (slavvco) [reply public/amls.css]: agreed to leave as it is, because ID's are used for acceptance tests

**Pull request merged**